### PR TITLE
Add prompt support for \D, \p and \_ options

### PR DESCRIPTION
**Pull request by mdsrosa**
This PR address the prompt support issues related in https://github.com/dbcli/mycli/issues/300.

I only add those 3 for now because those are the ones mentioned in the https://github.com/dbcli/mycli/issues/300. If the need for other options appears, I'll be happy to add them (or invite the user to add them).

Reviewers: @tsroten @j-bennet

### CODE REVIEW

**Comment by dbolser-ebi** [comment]
Awesome, many thanks for this!

May I ask that the prompt setting be read from the [mysql] section of the config, rather than the [client] section of the config (or even a [mycli] section?), as it messes with other client utilities such as mysqlcheck and friends.

Should this be a different ticket?

File: mycli/main.py [hunk 1]
```diff
@@ -693,6 +694,9 @@ def get_prompt(self, string):
         string = string.replace('\\d', sqlexecute.dbname or '(none)')
         string = string.replace('\\t', sqlexecute.server_type()[0] or 'mycli')
         string = string.replace('\\n', "\n")
+        string = string.replace('\\D', datetime.now().strftime('%a %b %d %H:%M:%S %Y'))
+        string = string.replace('\\p', socket.gethostbyname(socket.gethostname()))
```
Comment (mdsrosa) [inline mycli/main.py]: @amjith You're correct. I misread the MySQL documentation. Stopped at "The current TCP/IP" 😆 . Thanks! Will fix this.

**Comment by mdsrosa** [comment]
@dbolser-ebi 

Sorry if I misunderstood this, but do you mean that the `prompt` config should be read from the `my.cnf` instead of the `myclirc` file? If so, that sounds reasonable. And yes that would be another ticket. Would you mind creating one requesting that change, please?

**Comment by tsroten** [comment]
@dbolser-ebi Reading the prompt from `my.cnf` is definitely doable, but we wouldn't want to read it from the `[mysql]` section. That section is supposed to only be for the official MySQL command-line client's use. Third party clients, like mycli, are supposed to use the `[client]` section. Mycli-specific options generally live in `.myclirc`.

**Comment by amjith** [comment]
We currently read the ~/.my.cnf file but only honor the `[client]` section. If the prompt is defined in both ~/.my.cnf as well as ~/.myclirc we give preference to ~/.my.cnf. 

I think we should only honor the [client] sub-section because that's what clients are supposed to use.

**Comment by mdsrosa** [comment]
I'm merging this PR. 

@dbolser-ebi Do you have any other questions about this? If so, please, feel free to send a message to [our Gitter channel](https://gitter.im/dbcli/mycli). Thank you!

**Comment by dbolser-ebi** [comment]
Many thanks for this @mdsrosa ! How do I get the master version via pip?

**Comment by j-bennet** [comment]
`pip install -U https://github.com/dbcli/mycli/archive/master.zip`

**Comment by dbolser-ebi** [comment]
Hey! Speedy reply! Thanks so much!

**Comment by dbolser-ebi** [comment]
Please close #300 now if you like?

**Pull request merged**